one_fm/one_fm/hr_utils.py

```diff
@@ -1,130 +0,0 @@
-    if total_working_year < 5 or (total_working_year == 5 and total_working_days == 5*365):
-        #15 days salary is divided over a year and  that becomes each day's allocation.
-        total_amount = 15 * per_day_amount / 365 * total_working_days 
-    
-    elif total_working_year >= 5 and total_working_days > 5*365:
-        #calculation takes 15 days salary for 5 years and 30 days salary after 5 years 
-        amount_1 = 15 * per_day_amount / 365 * five_year_in_days
-        amount_2 = 30 * per_day_amount / 365 * (total_working_days-five_year_in_days)
-        total_amount = amount_1+amount_2
-    return total_amount
-
-def get_per_day_indemnity_amount(date_of_joining, to_date, indemnity_amount):
-    """To Calculate indemnity of the employee per day distributed across one year. 
-    This allows to get the per day calculation to be allocated every day.
-
-    Args:
-        date_of_joining ([date]): Employee's Joining Date
-        to_date ([data]): up until date
-        indemnity_amount ([currency]): Indemnity Amount from Salary Structure Assignment
-
-    Returns:
-        [type]: [description]
-    """
-    total_working_year = relativedelta(to_date, date_of_joining).years
-    total_working_days = (to_date - date_of_joining).days
-    per_day_amount = indemnity_amount/26
-
-    #calculate indemnity per day.
-    if total_working_year < 5:
-        return 15 * per_day_amount / 365
-    elif total_working_year >= 5 and total_working_days > (5*365):
-        return 30 * per_day_amount / 365 
-
-def allocate_daily_indemnity():
-    # Get List of Indemnity Allocation for today
-    allocation_list = frappe.get_all("Indemnity Allocation", filters={"expired": ["!=", 1]}, fields=["name"])
-    for alloc in allocation_list:
-        allow_allocation = True
-        allocation = frappe.get_doc('Indemnity Allocation', alloc.name)
-        date_of_joining = frappe.get_value("Employee",{"name":allocation.employee},["date_of_joining"])
-        indemnity_amount = frappe.get_value("Salary Structure Assignment",{"employee":allocation.employee},["indemnity_amount"])
-
-        # Set Daily Allocation
-        allocation.new_indemnity_allocated = get_per_day_indemnity_amount(date_of_joining, getdate(nowdate()) , indemnity_amount)
-        allocation.total_indemnity_allocated = allocation.total_indemnity_allocated+allocation.new_indemnity_allocated
-        allocation.save()
-
-def validate_leave_proof_document_requirement(doc, method):
-    '''
-        Function to validate Is Proof Document Required Flag in Leave Application
-        Triger form Validate hook of Leave Application
-    '''
-
-    if doc.leave_type and doc.status in ['Open', 'Approved']:
-        doc.is_proof_document_required = frappe.db.get_value('Leave Type', doc.leave_type, 'is_proof_document_required')
-        if doc.is_proof_document_required and not doc.proof_document:
-            frappe.throw(_("Proof Document Required for {0} Leave Type.!".format(doc.leave_type)))
```

Comment:
This method is not purely related to indemnity, It is related to the Leave Type and Leave Application

Reply:
Okay.